src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -45,39 +45,88 @@ namespace op {
 struct nonmaxsuppression
 {
     bool center_point_box = false;
+    bool use_dynamic      = false;
 
     template <class Self, class F>
     static auto reflect(Self& self, F f)
     {
-        return pack(f(self.center_point_box, "center_point_box"));
+        return pack(f(self.center_point_box, "center_point_box"),
+                    f(self.use_dynamic, "use_dynamic"));
     }
 
     std::string name() const { return "nonmaxsuppression"; }
 
     shape compute_shape(std::vector<shape> inputs) const
     {
         // requires at least 2 inputs
-        check_shapes{{inputs.at(0), inputs.at(1)}, *this}.only_dims(3);
-        auto lens = inputs.front().lens();
+        check_shapes{{inputs.at(0), inputs.at(1)}, *this, true}.only_dims(3).same_ndims();
```

Comment:
`use_dynamic` refers to the output of NMS being a dynamic shape. Looking at it again, there's a possibility that someone could pass a dynamic shape input without `use_dynamic` set to true. I'm not sure what exactly we would want to occur in that situation. However, with how `check_shapes` currently works it would state that NMS does not support dynamic shapes if we do the check without the `true` parameter.